wallet/trezor-common/src/lib.rs

```diff
@@ -0,0 +1,404 @@
+    pub media_uri: parity_scale_codec::alloc::vec::Vec<u8>,
+    pub media_hash: parity_scale_codec::alloc::vec::Vec<u8>,
+}
+
+#[derive(Encode)]
+pub struct OrderData {
+    /// The key that can authorize conclusion of an order
+    pub conclude_key: Destination,
+    /// `Ask` and `give` fields represent amounts of currencies
+    /// that an order maker wants to exchange.
+    /// E.g. Creator of an order asks for 5 coins and gives 10 tokens in
+    /// exchange.
+    pub ask: OutputValue,
+    pub give: OutputValue,
+}
+
+#[derive(Encode)]
+pub enum TxOutput {
+    /// Transfer an output, giving the provided Destination the authority to
+    /// spend it (no conditions)
+    #[codec(index = 0)]
+    Transfer(OutputValue, Destination),
+    /// Same as Transfer, but with the condition that an output can only be
+    /// specified after some point in time.
+    #[codec(index = 1)]
+    LockThenTransfer(OutputValue, Destination, OutputTimeLock),
+    /// Burn an amount (whether coin or token)
+    #[codec(index = 2)]
+    Burn(OutputValue),
+    /// Output type that is used to create a stake pool
+    #[codec(index = 3)]
+    CreateStakePool(H256, StakePoolData),
+    /// Output type that represents spending of a stake pool output in a block
+    /// reward in order to produce a block
+    #[codec(index = 4)]
+    ProduceBlockFromStake(Destination, H256),
+    /// Create a delegation; takes the owner destination (address authorized to
+    /// withdraw from the delegation) and a pool id
+    #[codec(index = 5)]
+    CreateDelegationId(Destination, H256),
+    /// Transfer an amount to a delegation that was previously created for
+    /// staking
+    #[codec(index = 6)]
+    DelegateStaking(Amount, H256),
+    #[codec(index = 7)]
+    IssueFungibleToken(TokenIssuance),
+    #[codec(index = 8)]
+    IssueNft(H256, NftIssuance, Destination),
+    #[codec(index = 9)]
+    DataDeposit(parity_scale_codec::alloc::vec::Vec<u8>),
+    #[codec(index = 10)]
+    Htlc(OutputValue, HashedTimelockContract),
+    #[codec(index = 11)]
+    CreateOrder(OrderData),
+}
+
+#[derive(Encode)]
+pub struct HashedTimelockContract {
+    // can be spent either by a specific address that knows the secret
+    pub secret_hash: HtlcSecretHash,
+    pub spend_key: Destination,
+
+    // or by a multisig after timelock expires making it possible to refund
+    pub refund_timelock: OutputTimeLock,
+    pub refund_key: Destination,
+}
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Debug, Encode, Decode)]
+pub struct H256(pub [u8; 32]);
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Copy, Debug, Encode, Decode)]
+pub struct HtlcSecretHash(pub [u8; 20]);
+
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode, Ord, PartialOrd)]
+pub enum OutPointSourceId {
+    #[codec(index = 0)]
+    Transaction(H256),
+    #[codec(index = 1)]
+    BlockReward(H256),
+}
+
+#[derive(FromPrimitive)]
+pub enum OutPointSourceIdIndex {
+    Transaction = 0,
+    BlockReward = 1,
+}
```

Comment:
As I can see, this is similar to AccountCommandIndex, but the difference is that there is already MintlayerUtxoType in the protobuf config which this enum should be consistent with. Remove it and use MintlayerUtxoType instead.